File: 686. Repeated String Match/_686.py
class Solution:
    def repeatedStringMatch(self, A: str, B: str) -> int:
        flag = 1
        while flag*len(A)<len(B):
            flag += 1
        if B in (flag*A):
            return flag
        flag += 1
        if B in (flag*A):
            return flag
        return -1

    def repeatedStringMatch1(self, A, B):
        t = -(-len(B) // len(A)) # Equal to ceil(len(b) / len(a))
        return t * (B in A * t) or (t + 1) * (B in A * (t + 1)) or -1

a = Solution()
print(a.repeatedStringMatch(A = "abab" , B = "aba"))

File: 686. Repeated String Match/686.go
package leetcode

import(
	"strings"
)

func repeatedStringMatch(A string, B string) int {
	flag := 1
	tmp := A
	for len(A)*flag < len(B){
		flag++
	}
	for i:=1;i<flag;i++{
		A += tmp
	}
	if strings.Contains(A, B){
		return flag
	}
	A += tmp
	if strings.Contains(A, B){
		return flag+1
	}
	return -1
}

func repeatedStringMatch1(A string, B string) int {
	max := len(B)/len(A) + 2
	n := len(B) / len(A)
	for n <= max {
		str := strings.Repeat(A, n)
		if strings.Contains(str, B) {
			return n
		}
		n++
	}

	return -1
}